### ISSUE TITLE
False Positive E3002 when using Fn::Transform

### ISSUE THREAD

**Issue by AV-IO** [aws-cloudformation/cfn-lint#2887]
### CloudFormation Lint Version

0.80.2

### What operating system are you using?

Mac

### Describe the bug

When calling `Fn::Transform` to return a list to a property, cfn-lint states that there is an E3002 error `Property is an object instead of List at ...`, but this does not take into consideration the multiple return types `Fn::Transform` has.

### Expected behavior

No error should be shown when a List is expected, but an object is provided when that object is a call to `Fn::Transform`.

### Reproduction template

```json
{
  "AWSTemplateFormatVersion": "2010-09-09",
  "Parameters": {
    "Zone": {
      "Description": "The zone you want the records in",
      "Type": "AWS::Route53::HostedZone::Id"
    },
    "Names": {
      "Description": "The record names you want created",
      "Type": "CommaDelimitedList"
    }
  },
  "Resources": {
    "Records": {
      "Type": "AWS::Route53::RecordSetGroup",
      "Properties": {
        "HostedZoneId": {
          "Ref": "Zone"
        },
        "RecordSets": {
          "Fn::Transform": {
            "Name": "Some-Macro-Name",
            "Parameters": {
              "Names": {
                "Ref": "Names"
              }
            }
          }
        }
      }
    }
  }
}
```

**Comment by PatMyron** [comment]
tracking in https://github.com/aws-cloudformation/cfn-lint/issues/476
